### ISSUE TITLE
Comments: How to install Ollama AI Tool with Oterm and Open WebUI on NixOS

### ISSUE THREAD

**Issue by fictionbecomesfact** [fictionbecomesfact/fictionbecomesfact_comments#73]
Comments for [How to install Ollama AI Tool with Oterm and Open WebUI on NixOS](https://fictionbecomesfact.com/notes/nixos-ollama-oterm-openwebui)

**Comment by gelerum** [comment]
The option below shouldn't be used with `ports = ["127.0.0.1:3000:8080"]`, the option below give access to all ports in container, from host that's why you connect to WebUI using 127.0.0.1:8080. If you delete port below, you will be able to connect only using 127.0.0.1:3000. Or you can change hostPort in  `ports = ["127.0.0.1:<hostPort>:8080"]`.
```nix
extraOptions = [
    "--network=host"
  ];
```

Everything else was really helpful, thanks!

**Comment by mwmdev** [comment]
Have you found a way to make ollama environment variables work? I'm trying to have models stored on an external drive.

**Comment by gelerum** [comment]
> Have you found a way to make ollama environment variables work? I'm trying to have models stored on an external drive.

No, but you can create a symlink from your external drive directory to /var/lib/private/ollama/models

**Comment by tonybeyond** [comment]
Thanks for this
I've managed to have the environment variable working like this:

``` nix
 services.ollama = {
    enable = true;
    acceleration = "cuda";
    openFirewall = true;    # Ensure the firewall allows incoming traffic
};

systemd.services.ollama.serviceConfig = {
  Environment = [ "OLLAMA_HOST=0.0.0.0:11434" ];
};
```

**Comment by fictionbecomesfact** [comment]
Thank you @tonybeyond for sharing, I have incorporated it into the note
